scripts/code_style.py

```diff
@@ -119,13 +119,31 @@ def check_style_is_correct(src_file_list: List[str]) -> bool:
 
         # Uncrustify makes changes to the code and places the result in a new
         # file with the extension ".uncrustify". To get the changes (if any)
-        # simply diff the 2 files.
-        diff_cmd = ["diff", "-u", src_file, src_file + ".uncrustify"]
-        cp = subprocess.run(diff_cmd, check=False)
+        # simply diff the 2 files
+        if use_git_diff:
+            cmp_cmd = ["cmp", "-s", src_file, src_file + ".uncrustify"]
+        else:
+            cmp_cmd = ["diff", "-u", src_file, src_file + ".uncrustify"]
+            
+        cp = subprocess.run(cmp_cmd, check=False)
 
         if cp.returncode == 1:
-            print(src_file + " changed - code style is incorrect.")
             style_correct = False
+            
+            if use_git_diff:
+                if use_pager:
+                    diff_cmd = ["git", "diff", "-u", src_file, src_file + ".uncrustify"]
```

Comment:
If someone somehow checks in the `.uncrustify`, this will diff that file with its checked-in version. Use `--no-index` to avoid this surprising edge case.